packages/trade-widget/src/components/BuyWidget/BuyWidget.tsx

```diff
@@ -0,0 +1,207 @@
+import { SignedOrder, ZeroEx } from '0x.js';
+import { InjectedWeb3Subprovider, RedundantRPCSubprovider } from '@0xproject/subproviders';
+import { AbiDecoder, BigNumber } from '@0xproject/utils';
+import { Web3Wrapper } from '@0xproject/web3-wrapper';
+import {
+    Box,
+    Button,
+    Card,
+    CardContent,
+    CardHeaderTitle,
+    Column,
+    Columns,
+    Container,
+    Content,
+    Control,
+    Field,
+    Image,
+    Input,
+    Label,
+    Select,
+    TextArea,
+} from 'bloomer';
+import 'bulma/css/bulma.css';
+import * as _ from 'lodash';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import * as Web3 from 'web3';
+import * as Web3ProviderEngine from 'web3-provider-engine';
+
+import { artifacts } from '../../artifacts';
+import { ForwarderWrapper } from '../../contract_wrappers/forwarder_wrapper';
+
+import { Dispatcher } from '../../redux/dispatcher';
+import {
+    AccountTokenBalances,
+    AccountWeiBalances,
+    AssetToken,
+    Quote,
+    QuoteRequest,
+    TokenBalances,
+    TokenPair,
+} from '../../types';
+import AccountBlockie from '../AccountBlockie';
+import TokenSelector from '../TokenSelector';
+
+/* tslint:disable:prefer-function-over-method member-access */
+
+interface BuyWidgetProps {
+    onTokenChange?: (token: string) => any;
+    onAmountChange?: (amount: BigNumber) => any;
+    address: string;
+    networkId: number;
+    weiBalances: AccountWeiBalances;
+    tokenBalances: AccountTokenBalances;
+    selectedToken: AssetToken;
+    web3Wrapper: Web3Wrapper;
+    zeroEx: ZeroEx;
+    dispatcher: Dispatcher;
+    quote: Quote;
+}
+
+interface BuyWidgetState {
+    amount?: BigNumber;
+}
+
+const ETH_DECIMAL_PLACES = 18;
+class BuyWidget extends React.Component<BuyWidgetProps, BuyWidgetState> {
+    private _forwarder: ForwarderWrapper;
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            amount: new BigNumber(1),
```

Comment:
Do we want this to default to `1`? Also, rename `amount`. Amount is base units? units? Unclear.